types/react-mentions/index.d.ts

```diff
@@ -41,21 +42,7 @@ export interface MentionsInputProps {
     style?: any;
     regex?: RegExp;
     suggestionsPortalHost?: Element;
-}
-
-/**
- * Exposes the type for use with the @see MentionsInputComponent.wrappedInstance which is added by react-mentions' use of substyle (https://github.com/jfschwarz/substyle).
- */
-export interface MentionsInputComponentUnrwapped extends React.Component<MentionsInputProps> {
-    inputRef?: HTMLInputElement | HTMLTextAreaElement;
-}
-
-/**
- * Used with @see React.RefObject<MentionsInputComponent>.
- */
-export interface MentionsInputComponent extends React.Component<MentionsInputProps> {
-    // MentionsInput uses substyle (https://github.com/jfschwarz/substyle) which adds this wrappedInstance
-    wrappedInstance?: MentionsInputComponentUnrwapped;
```

Comment:
Hey @efedorenko! Glad to know someone else is using these types! In this specific case, why are you changing this? I'm using the wrappedInstance in my app as I found it was the only way to get to the underlying inputRef. I'm doing this in a code block as follows:

```
<MentionsInput
   ref={this.attachRefs}
   ...
 >
```

Then later in `attachRefs`:

```
private attachRefs = (ref: MentionsInputComponent | null): void => {
        if (!ref) {
            return;
        }
        const wrappedInstance = ref.wrappedInstance;
        if (!wrappedInstance) {
            return;
        }
        this.attachInputRef(wrappedInstance);
};
```

The above code is working for me, but I accept maybe I'm missing something (or maybe I'm using a different version). Can you help me understand why you're making this change and what you're seeing?

Also a couple things I think we should follow up on:

1. If we do end up making this change, it is a [incompatible API change according to semver](https://semver.org/) so I think we'll have to bump the MAJOR version.
2. We should [update tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/types/react-mentions/react-mentions-tests.tsx) to touch `inputRef` so we

Reply:
Just noticed you're from wildbit! You guys rock!

Reply:
Hey @activescott! Thanks for sharing these types, it made trying and using the library so much easier!

I made the change only because react-mentions [was updated to v2.4](https://github.com/signavio/react-mentions/compare/v2.3.1...v2.4.0) and they added `inputRef` property to MentionsInput, so there is no need for this workaround anymore. A regular React.RefObject can be passed to MentionsInput now, which makes things much easier.

I am new to this, so please bear with me if it doesn't make sense, but react-mention released this change with a minor version bump so I just followed their lead. I feel like it's a general practice for DefinitelyTyped typings and original projects to have matching major and minor version numbers.

I'll update tests in a little bit, thanks for the reminder.

P.S. So happy to see that you know Wildbit! Do you use any of our products?

Reply:
Updated tests. BTW, it seems like your comment was cut off in the middle of 2nd point.

Reply:
@efedorenko Thanks for the update and the pointer to react-mention 2.4's addition of the top-level `inputRef` prop. While it is obvious that we should all be using the top-level `inputRef` prop rather than poking around into `wrappedInstance` (like me 🤦‍♂️), I think just removing the types during a minor version change is too heavy handed for a couple reasons:
1. _Removing_ `wrappedInstance` and `MentionsInputComponentUnrwapped.inputRef` in the types during a minor semver change will introduce a breaking change for @types/react-mentions consumers that are already using it. Anyone using the default [caret range](https://github.com/npm/node-semver#caret-ranges-123-025-004) for @types/react-mentions dependencies will be automatically upgraded to the 2.4.0 types and it will suddenly break their build.
2. AFAICT (haven't tested) the underlying code for react-mentions even in v2.4.0+ appears to still support the underlying `wrappedInstance` and `MentionsInputComponentUnrwapped.inputRef`. So I think the types are a faithful representation of the underlying code. 

With that said, to guide people to the new inputRef prop, I suggest you add a [JSDoc `@deprecated` comment](http://usejsdoc.org/tags-deprecated.html) to the `MentionsInputComponentUnrwapped.inputRef` member with a comment pointing them to use the new top-level inputRef prop. This will cause linters and users viewing comments in editors to go in the right direction, but not break their builds. I'm thinking something like the following:
```
/**
 * Exposes the type for use with the @see MentionsInputComponent.wrappedInstance which is added by react-mentions' use of substyle (https://github.com/jfschwarz/substyle).
 */
export interface MentionsInputComponentUnrwapped extends React.Component<MentionsInputProps> {
    /**
     * @deprecated since version 2.4.0. Please use @see MentionsInputProps.inputRef
     */
    inputRef?: HTMLInputElement | HTMLTextAreaElement;
}
```


Let me know if you have a different POV or would like a hand.

Reply:
That sounds like a good solution. I reverted removal of these types and added `@deprecated` comment. Let me know if there is anything else you think should be done!

Reply:
LGTM. :shipit: